### PR TITLE
Assignment 2 by Meijuan Zeng

### PR DESCRIPTION
**Pull request by tomato018**

### CODE REVIEW

File: Meijuan Zeng-assignment 2.Rmd [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+total <- bind_rows(total_st_EDGE, total_t_EDGE)
+VERTEX <- full_join(VERTEX, total, by="id")
+# Graph
+my_gr <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+plot(my_gr,layout=layout.fruchterman.reingold, 
+     vertex.color=as.factor(VERTEX$gender), 
+     edge.arrow.size = 0.5,
+     edge.width=EDGE$count,
+     vertex.size = VERTEX$count*2)
+```
```
Comment (DaveCody) [inline Meijuan Zeng-assignment 2.Rmd]: Perfect! Nice work combining the teacher and student counts and joining the vertex and join dataframes.

File: Meijuan Zeng-assignment 2.Rmd [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+#select name and classes from d1: 
+d2 <- select(d1, "Q8": "Q7") %>% unite(name, Q8, Q9, sep = " ") 
+class_name <- c("Q1 : Q7")
+```
+
+```{r}
+#Gather name by the first and last names,all the classes into class_name, arrange their orders and remove the empty rows:
+d3 <- gather(d2, class, class_name, 2: 7, na.rm = TRUE) %>%  select(name, class_name)%>%arrange(name,class_name)%>%na.omit()%>%filter(class_name>0, class_name != "HUDK 4050")
+
```
Comment (DaveCody) [inline Meijuan Zeng-assignment 2.Rmd]: If we look at the class names in the data set, not all of the formatting is standardized. Some of the class names include spaces and others don't. For example, the code will treat HUDK 4050 and HUDK4050 as two different classes, but they should be the same. You can use gsub to remove the spaces and standardize the format.

File: Meijuan Zeng-assignment 2.Rmd [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+library(igraph)
+g <- graph.adjacency(d6, mode = "undirected", weighted = TRUE)
+plot(g,layout = layout.fruchterman.reingold, edge.arrow.size = 0.5, vertex.label.size = 5)
+
+#centrality
+degree(g)    
+
+#betweeness centrality
+betweenness(g)
+```
```
Comment (DaveCody) [inline Meijuan Zeng-assignment 2.Rmd]: What do the centrality and betweenness metrics mean? There is some info in the slides that can help with the synthesis.

File: Meijuan Zeng-assignment 2.Rmd (hunk 2 above)
Comment (tomato018) [reply Meijuan Zeng-assignment 2.Rmd]: Hi I just add gsub in my new file.

**Pull request closed without merging**